extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -165,12 +180,22 @@ class ConnectorsManager {
                     kafkaConnectHost: this._kafkaConnectHost,
                     kafkaConnectPort: this._kafkaConnectPort,
                 });
+                if (buckets.length > this._maximumBucketsPerConnector) {
+                    bucketsExceedingLimit += buckets.length - this._maximumBucketsPerConnector;
+                    this._logger.warn('Connector has more bucket than expected', {
+                        method: 'ConnectorsManager._getOldConnectors',
+                        connector: connector.name,
+                        numberOfBuckets: buckets.length,
+                        allowed: this._maximumBucketsPerConnector,
+                    });
+                }
                 this._logger.debug('Successfully retreived old connector', {
                     method: 'ConnectorsManager._getOldConnectors',
                     connector: connector.name
                 });
                 return connector;
             }));
+            this.emit('connectors-reconciled', bucketsExceedingLimit);
```

Comment:
`_getOldConnectors` is only called once on startup, from `initializeConnectors()`.

the actual reconcilization is done in `_updateConnectors()`

--> the event must be called at the end of `_updateConnectors()`, and maybe (but not sure it is strictly needed?) at the end of `initializeConnectors()`

Reply:
done, also adapted the logic